File: src/completed/java/euler0002.java
/**
 * Created by Holden on 11/19/2016. - COMPLETED
 *
 * Problem 2 - Even Fibonacci Numbers
 *
 */

public class Prob0002 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        int answer = evenFibonacciNumbers(4000000);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static int evenFibonacciNumbers(int upperBound) {
        int fib1 = 1;
        int fib2 = 2;
        int temp;
        int fibSum = 0;

        while (fib2 < upperBound) {
            if (fib2 % 2 == 0) {
                fibSum += fib2;
            }
            temp = fib2;
            fib2 += fib1;
            fib1 = temp;
        }

        return fibSum;
    }
}

File: python/completed/euler0002.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 12/17/2015
#
# Problem 2 - Even Fibonacci Numbers
#
###

def even_fibonacci(upper_bound):
    f1 = 1
    f2 = 2
    fib_sum = 0
    
    while f2 < upper_bound:
        if f2 % 2 == 0:
            fib_sum += f2
        place = f2
        f2 += f1
        f1 = place
        
    return fib_sum